src/CredentialSource/AwsNativeSource.php

```diff
@@ -328,6 +328,21 @@ public static function getSigningVarsFromEnv(): ?array
         return null;
     }
 
+    /**
+     * Gets the unique key for caching
+     * For AwsNativeSource the values are:
+     * imdsv2SessionTokenUrl:securityCredentialsUrl:regionUrl:regionalCredVerificationUrl:audience
+     *
+     * @return string
+     */
+    public function getCacheKey(): string
+    {
+        return $this->imdsv2SessionTokenUrl .
+            $this->securityCredentialsUrl .
```

Comment:
Since these first two can be `null`, we should only add them if they exist (similar to `sub` in `ServiceAccountCredentials::getCacheKey`)

Also, I think we should concatenate them with a `.` for consistency and readability.

Reply:
You can concatenate null to a string in PHP and will not affect the result. If we want to do it just for being verbose? I have my reservations but I could see why:
```
$a = null;
$b = null;
$c = 'Hector';
$d = 'Mendoza';

$test = $a . $b . $c . $d;
print($test . "\n");
```
Result:
![image](https://github.com/googleapis/google-auth-library-php/assets/9062626/b2f601e5-f73d-4f2b-a6ba-bd92be91182c)

> Also, I think we should concatenate them with a . for consistency and readability.
My apologies, I do not understand this comment. To my understanding I am already concatenating everything with `.` unless I am misunderstanding something?

Reply:
AH! Sorry, now I understand what you mean, we meant the character `.` not the operator `.` haha. I went ahead and removed all the `.` as we strip them anyway and keep everything consistent.

Reply:
Yes! I meant let's concactinate with a `.`. This will also make the string easier to demonstrate in the PHPDoc.

Also, while you can concatinate `null`, and PHP will cast to an empty string, I'd prefer to not use implicit casting, and treat nulls as nulls and not as empty strings.